src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -50,27 +225,296 @@ def execute(self, reqStatus):
+            self.logger.info(msg)
+            self.updateReportDict(summary, "total_num_requests", total_num_requests)
+            self.updateReportDict(summary, "clean_num_requests", clean_num_requests)
+            self.updateReportDict(summary, "archived_num_requests", archived_num_requests)
+        except Exception as ex:
+            msg = "{}: Unknown exception while running MSRuleCleaner thread. ".format(self.currThreadIdent)
+            msg += "Error: {}".format(str(ex))
+            self.logger.exception(msg)
+            self.updateReportDict(summary, "error", msg)
+
         return summary
 
+    def _execute(self, reqRecords):
+        """
+        Executes the MSRuleCleaner pipelines based on the workflow status
+        :param reqList: A list of ReqManager records to work on
+        :return:        number of properly cleaned requests
+        """
+        # NOTE: We iterate destructively through the reqList
+        # NOTE: The Input Cleanup, the Block Level Cleanup and the Archival
+        #       Pipelines are executed sequentially in the above order.
+        #       This way we assure ourselves that we archive only workflows
+        #       that have accomplished the needed cleanup
+
+        clean_num_requests = 0
+        total_num_requests = 0
+
+        # Call the workflow dispatcher:
+        for reqName, req in reqRecords.items():
+            wflow = MSRuleCleanerWflow(reqRecords.pop(reqName))
+            self._dispatchWflow(wflow)
+            msg = "\n----------------------------------------------------------"
+            msg += "\nMSRuleCleanerWflow: %s"
+            msg += "\n----------------------------------------------------------"
+            self.logger.debug(msg, pformat(wflow))
+            total_num_requests += 1
+            if self._checkClean(wflow):
+                clean_num_requests += 1
+
+        # Report the counters:
+        for pline in self.cleanuplines:
+            msg = "%s: Workflows cleaned by pipeline: %s: %d"
+            self.logger.info(msg, self.currThreadIdent, pline.name, self.wfCounters['cleaned'][pline.name])
+        archived_num_requests = self.wfCounters['archived']
+        msg = "%s: Workflows archived: %d"
+        self.logger.info(msg, self.currThreadIdent, self.wfCounters['archived'])
+        return clean_num_requests, total_num_requests, archived_num_requests
+
+    def _dispatchWflow(self, wflow, *args, **kwargs):
+        """
+        A function intended to dispatch a workflow (e.g based on its status)
+        through one or more functional pipelines in case there is some more
+        complicated logic involved in the order we execute them but not just
+        a sequentially
+        """
+        # NOTE: The following dispatch logic is a subject to be changed at any time
+
+        if wflow['RequestType'] == 'Resubmission':
```

Comment:
Agreed, since `_execute()` is taking care of splitting the work in smaller junks all filters applied should definitely go in here instead of the dispatcher. And actually it makes the `_dispatcherWflow()` method look much more structured.

Done.